---
output:
  html_document:
    df_print: paged
    fig.align: center
    self_contained: yes 
    fig.height: 4
    fig.width: 8
    theme: united
    toc: yes
    toc_depth: 4
    toc_float: yes
    number_sections: yes
    code_folding: hide
title: "Auswertung TC Daten"
author: "David Lilek"
date: "`r format(Sys.time(), '%d %B %Y, %X')`"
editor_options: 
  markdown: 
    wrap: 72
---



# Load Libraries & Plot function & Colors

```{r message=FALSE, warning=FALSE}
# load libaries
library(ggplot2)
library(tidyverse)
library(dplyr)
library(UpSetR)
library(knitr)
library(reshape2)
library(eulerr)
library(UniprotR)
library(stringr)
library(ggfittext)
```

```{r}
# load functions for proportional venn diagrams
euler_venn_2comp <- function(lst, 
                             label_venn,
                             color_venn, 
                             cex_venn,
                             grid,
                             gList,
                             h_just,
                             v_just) {
  p<-plot(euler(lst),
          input="union",
          shape="ellipse",
          quantities=list(type = c("percent", 'counts'),
                          cex = cex_venn), #font size for A, B and intersection
          key=TRUE,
          labels=FALSE,
          lty = c(1,1),
          legend = list(labels = label_venn,
                        side = "bottom",
                        nrow = 3, ncol = 1),
          edges=TRUE,
          fills = color_venn, #color for A,B and intersection 
          alpha=1)
  # adjust label position
  # https://stackoverflow.com/questions/75177293/how-to-change-the-label-position-when-plotting-venn-diagram-from-eulerr-package
  tags <- p$children[[2]]$children[[1]]$children$tags$children
  tags <- do.call(grid::gList, lapply(tags, function(x) {
    x$children[[2]]$label <- sub(" \\(", "\n(", x$children[[2]]$label)
    x$children[[2]]$just <- NULL
    x$children[[2]]$hjust <- h_just
    x$children[[2]]$vjust <- v_just
    x}))
  
  p$children[[2]]$children[[1]]$children$tags$children <- tags
  print(p)
}


euler_venn_3comp <- function(lst, cex_venn, label_venn, color_venn, grid, gList, h_just, v_just) {
  ##################################
  ###### venn diagram 3 comparisons
  ##################################
  p<-plot(euler(lst),
          input="union",
          shape="ellipse",
          quantities=list(type = c("percent", 'counts'),
                          cex = cex_venn), #font size for A, B, C and intersection
          key=TRUE,
          labels=FALSE,
          lty = c(1,1,1),
          legend = list(labels = label_venn,
                        side = "bottom",
                        nrow = 3, ncol = 1),
          edges=TRUE,
          fills = color_venn, #color for A,B and intersection 
          alpha=1)
  # adjust labels
  tags <- p$children[[2]]$children[[1]]$children$tags$children
  tags <- do.call(grid::gList, lapply(tags, function(x) {
    x$children[[2]]$label <- sub(" \\(", "\n(", x$children[[2]]$label)
    x$children[[2]]$just <- NULL
    x$children[[2]]$hjust <- h_just
    x$children[[2]]$vjust <- v_just
    x}))
  
  p$children[[2]]$children[[1]]$children$tags$children <- tags
  print(p)
}

```

```{r}
# load colors
#dark: total protein extract
#light: fractionation
col_X_dark <- rgb(204,0,0, maxColorValue = 255)
col_X_light <- rgb(255,100,100,maxColorValue = 255)
#fractions high MW SEC
col_X_lowMW  <- rgb(255,150,150,maxColorValue = 255)
#fractions low MW SEC
col_X_highMW <- rgb(255,230,230, maxColorValue = 255)
col_X_carbonate <- rgb(255,217,102, maxColorValue = 255)
col_X_citrate <- rgb(169,208,142,maxColorValue = 255)
# D2
col_Y_dark <- rgb(0,0,204, maxColorValue = 255)
col_Y_light <- rgb(100,100,255, maxColorValue = 255)

```

# Figure 2

## A - Buffer comparison

```{r message=FALSE, warning=FALSE, fig.width=4,fig.height=4}
# get concentration values and define group
values <- c(6.39,
            6.09,
            12.33,
            10.68,
            15.1,
            14.12)
buffer <- c("Citrate","Citrate","Phosphate","Phosphate","Carbonate","Carbonate")
data <- as.data.frame(cbind(values,buffer))

# perform anova, post-hoc test and plot summary
res <- aov(values ~ buffer, data = data)
summary(res)
TukeyHSD(res)

# create boxplot 
data_4plot <- as.data.frame(cbind(values, factor(buffer)))
ggplot(data_4plot, aes(x=reorder(buffer,values), y=values)) + 
  geom_boxplot() + 
  labs(x = "",
       y = "Protein concentration [mg/mL]") +
  scale_x_discrete(labels = c("Citrate" = "Citrate", "Phosphate" = "Phosphate", "Carbonate" = "Carbonate"))
```

## B - Venn diagram buffer comparison

```{r message=FALSE, warning=FALSE, fig.width=4,fig.height=4}
# read in filtered results of maxquant
res_raw <- readRDS(file = "results/mqpar_extracts_2gether_reanalysisCM_extraktejakob_proteinGroups.txt_Unique.RDS")
res_raw <- res_raw[[1]]
# get rid of the column with the FASTA headers
res <- res_raw[,-ncol(res_raw)]
# multiply by 1; TRUE = 1; FALSE = 0
res <- res*1
# combine all carbonate (CAR), citrate (CIT), and phosphate (PHOS) columns 
CAR <- apply(res[,c(1:2)], 1, any)
CIT <- apply(res[,c(3:4)],1,any)
PHOS <- apply(res[,c(5:6)],1,any)

# get primary accession ID from the FASTA headers
fasta <- c()
for (i in 1:length(res_raw$FASTA)){
  tmp <- strsplit(res_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}

# combine FASTA ID with merged samples and prepare data.frame for further data analysis
res_clear <- as.data.frame(cbind(CAR,CIT,PHOS,fasta))
res_clear$CAR <- as.logical(res_clear$CAR)
res_clear$CIT <- as.logical(res_clear$CIT)
res_clear$PHOS <- as.logical(res_clear$PHOS)
res_clear <- res_clear[,-4]*1
res_clear[res_clear==0] <- NA

# check for rows which are not NA
# column 1 belongs to CAR, column 2 to CIT and column 3 to PHOS
proteins <- as.data.frame(which(!is.na(res_clear), arr.ind=TRUE))
CAR <- proteins %>% dplyr::filter(col == 1)
CIT <- proteins %>% dplyr::filter(col == 2)
PHOS <- proteins %>% dplyr::filter(col == 3)
# combine to list
lst <- list("CIT"=as.double(CIT[,1]),
            "PHOS"=as.double(PHOS[,1]),
            "CAR"=as.double(CAR[,1]))
# proportional venn diagram
euler_venn_3comp(lst, label_venn = c("Citrate buffer pH 4.8", "Phosphate buffer pH 8","Carbonate buffer pH 10.8"),
                 color_venn = c(col_X_citrate,col_X_dark,col_X_carbonate),
                 cex_venn = c(0.85,1,0.5,0.75,0,1,1), 
                 h_just = 0.5, v_just = 0.5)

# print results
print(c("car","cit","phos"))
sum(res_clear$CAR, na.rm = TRUE)
sum(res_clear$CIT,na.rm = TRUE)
sum(res_clear$PHOS,na.rm = TRUE)
```


# Table 1 - Comparison noTreatment, Heatshock, ProteaseInhibitor

```{r, fig.width=4,fig.height=4}
# read in filtered results of maxquant 
data <- read.csv("results_2gether.csv", sep=";")
# use 2 unique peptides and no MBR
data_sub <- as.data.frame(data$Unique.no.MBR.2.peptides)
# the first 18 rows (6 replicates for each treatment) belong to this comparison
data_sub <- as.data.frame(data_sub[c(1:18),])
data_sub$group <- c(rep("No treatment",6),
                   rep("Heatshock",6),
                   rep("Protease inhibitor",6))
colnames(data_sub) <- c("number","group")
print("Median and sd no Treatment")
median(subset(data_sub,group=="No treatment")[,1])
sd(subset(data_sub,group=="No treatment")[,1])
print("Median and sd Heatshock")
median(subset(data_sub,group=="Heatshock")[,1])
sd(subset(data_sub,group=="Heatshock")[,1])
print("Median and sd ProteaseInhibitor")
median(subset(data_sub,group=="Protease inhibitor")[,1])
sd(subset(data_sub,group=="Protease inhibitor")[,1])
# perform ANOVA, post-hoc test and plot summary
anova <- aov(number ~ group, data=data_sub)
summary(anova)
Tukey <- TukeyHSD(anova)
plot(Tukey)
Tukey
```

# Figure 3

## A - D1/D2 comparison

```{r, fig.width=4,fig.height=4}
# D1-D2 comparison
# prepare data D1
D1_raw <- readRDS(file = "results/results_run1_mqpar_extracts_2gether_E4_D1_a-j_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
D1_raw <- D1_raw[[1]]
D1 <- dplyr::select(D1_raw, contains("Pool"))
D1 <- apply(D1, 1, any)
fasta <- c()
for (i in 1:length(D1_raw$FASTA)){
  tmp <- strsplit(D1_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
D1 <- as.data.frame(cbind(D1, fasta))

# prepare data D2
D2_raw <- readRDS(file = "results/results_run1_mqpar_extracts_2gether_D2_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
D2_raw <- D2_raw[[1]]
D2 <- dplyr::select(D2_raw, contains("Gesamt"))
D2 <- apply(D2, 1, any)
fasta <- c()
for (i in 1:length(D2_raw$FASTA)){
  tmp <- strsplit(D2_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
D2 <- as.data.frame(cbind(D2, fasta))

# merge D1 & D2 and prepare data.frame for plotting
D1_D2_comparison  <- merge(D1, D2,
                           all = TRUE,
                           by = "fasta")

D1_D2_comparison$D1 <- as.logical(D1_D2_comparison$D1)
D1_D2_comparison$D2 <- as.logical(D1_D2_comparison$D2)
D1_D2_comparison <- D1_D2_comparison[,-1]*1
D1_D2_comparison[D1_D2_comparison==0] <- NA

proteins <- as.data.frame(which(!is.na(D1_D2_comparison), arr.ind=TRUE))

D1_inter <- proteins %>% dplyr::filter(col == 1)
D2_inter <- proteins %>% dplyr::filter(col == 2)

# combine to list
lst <- list("D1"=as.double(D1_inter[,1]),
            "D2"=as.double(D2_inter[,1]))

# plot results as venn diagram
euler_venn_2comp(lst, label_venn = c("D1: Phosphate buffer", "D2: Phosphate buffer with urea-triton X"), color_venn = c(col_X_dark, col_Y_dark), cex_venn = c(1,1), h_just = 0.5, v_just = 0.5)

# print summary
print(c("D1","D2"))
sum(D1_D2_comparison$D1,na.rm = TRUE)
sum(D1_D2_comparison$D2,na.rm = TRUE)
```


## B - D1/D1b/D2 comparison

```{r, fig.width=4,fig.height=4}
# load result for comparison and prepare data
data_raw <- readRDS("results/D5_extracts_Razor.RDS")
data <- data_raw[[1]]
data_clear <- dplyr::select(data, contains("_b"))
D1 <- data_clear$Razor...unique.peptides.1_b_01+ data_clear$Razor...unique.peptides.1_b_02
D1_b <- data_clear$Razor...unique.peptides.2_b_01 + data_clear$Razor...unique.peptides.2_b_02
D2 <- data_clear$Razor...unique.peptides.4_b_01 + data_clear$Razor...unique.peptides.4_b_02
data_clear <- as.data.frame(cbind(D1, D1_b, D2))
data_clear[data_clear==0] <- NA
data_clear[data_clear==2] <- 1

proteins <- as.data.frame(which(!is.na(data_clear), arr.ind=TRUE))
D1 <- proteins %>% dplyr::filter(col == 1)
D1_b <- proteins %>% dplyr::filter(col == 2)
D2 <- proteins %>% dplyr::filter(col == 3)

# combine to list
lst <- list(D1 = as.double(D1[,1]),
            D1_b = as.double(D1_b[,1]),
            D2 = as.double(D2[,1]))

# plot as proportional venn diagram
euler_venn_3comp(lst, label_venn = c("D1: Phosphate buffer",
                                     "D1b: Carbonate buffer",
                                     "D2: Phosphate buffer with urea-triton X"), 
                 color_venn = c(col_X_dark,
                                col_X_carbonate,
                                col_Y_dark), 
                 cex_venn = c(1,1,1,1,0,1,1),
                 h_just = 0.5, v_just = 0.5)

# print summary results
print(c("D1","D1_b","D2"))
sum(data_clear$Razor...unique.peptides.1_b_01,na.rm = TRUE)
sum(data_clear$Razor...unique.peptides.2_b_01,na.rm = TRUE)
sum(data_clear$Razor...unique.peptides.4_b_01,na.rm = TRUE)
```

## C - Comparing 2D/3D

```{r, fig.width=8,fig.height=4,message=FALSE, warning=FALSE}
# read in an prepare data for 2D approach
D1_raw <- readRDS(file = "results/results_run1_mqpar_extracts_2gether_E4_D1_a-j_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
D1_raw <- D1_raw[[1]]
D1 <- dplyr::select(D1_raw, contains("Pool"))
D1 <- apply(D1, 1, any)
#D1 <- as.data.frame(cbind(D1, D1_raw$FASTA))
fasta <- c()
for (i in 1:length(D1_raw$FASTA)){
  tmp <- strsplit(D1_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
D1 <- as.data.frame(cbind(D1, fasta))
D2_raw <- readRDS(file = "results/results_run1_mqpar_extracts_2gether_D2_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
D2_raw <- D2_raw[[1]]
D2 <- dplyr::select(D2_raw, contains("Gesamt"))
D2 <- apply(D2, 1, any)
fasta <- c()
for (i in 1:length(D2_raw$FASTA)){
  tmp <- strsplit(D2_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
D2 <- as.data.frame(cbind(D2, fasta))
D1_D2_comparison  <- merge(D1, D2,
                           all = TRUE,
                           by = "fasta")
D1_D2_comparison$comb <- apply(D1_D2_comparison[,c(2:3)], 1, any)

# read in an prepare data for 3D approach
data <- readRDS("results/D5_extracts_Razor.RDS")
data <- data[[1]]
b <- dplyr::select(data, contains("_b"))
b <- b[,-c(5:6)]
b <- as.data.frame(apply(b, 1, any))
fasta <- c()
for (i in 1:length(data$FASTA)){
  tmp <- strsplit(data$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
b$fasta <- fasta
colnames(b) <- c("b","fasta")

# merge 2D with 3D and prepare date for plotting
comparison_2D_3D  <- merge(D1_D2_comparison[,c(1,4)],b,
                           all = TRUE,
                           by = "fasta")
colnames(comparison_2D_3D) <- c("fasta","E4","E5")
comparison_2D_3D$D1 <- as.logical(comparison_2D_3D$E4)
comparison_2D_3D$D2 <- as.logical(comparison_2D_3D$E5)
comparison_2D_3D <- comparison_2D_3D[,-1]*1
comparison_2D_3D[comparison_2D_3D==0] <- NA

proteins <- as.data.frame(which(!is.na(comparison_2D_3D), arr.ind=TRUE))

comp_2D <- proteins %>% dplyr::filter(col == 1)
comp_3D <- proteins %>% dplyr::filter(col == 2)

# combine to list
lst <- list("2D"=as.double(comp_2D[,1]),
            "3D"=as.double(comp_3D[,1]))

# plot with upset R
upset(fromList(lst), 
      order.by = "freq",
      nsets = 10,
      nintersects = 15,
      keep.order = TRUE,
      sets=c("2D", "3D"),
      mainbar.y.label = "Number of shared proteins",
      sets.x.label = "Total number of proteins",
      set_size.show = TRUE,
      set_size.angles = 0,
      set_size.scale_max = 1500
)
```




# Figure 4

## A - Compare electrophoretic separation and SEC

```{r, fig.width=4,fig.height=4}
# read in and prepare data from low and high MW SEC
AF_TF_raw <- readRDS(file = "results/results_run1_mqpar_extracts_2gether_E4_D1_AF_TF_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
AF_TF_raw <- AF_TF_raw[[1]]
AF <- dplyr::select(AF_TF_raw, contains("AF"))
AF <- apply(AF, 1, any)
fasta <- c()
for (i in 1:length(AF_TF_raw$FASTA)){
  tmp <- strsplit(AF_TF_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
AF <- as.data.frame(cbind(AF, fasta))
TF <- dplyr::select(AF_TF_raw, contains("TF"))
TF <- apply(TF, 1, any)
TF <- as.data.frame(cbind(TF, fasta))
# read in and prepare data from electrophorectic separation
B_raw <- readRDS(file = "results/results_run1_mqpar_extracts_2gether_E4_D1_B_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
B_raw <- B_raw[[1]]
B <- dplyr::select(B_raw, contains("Pool_G"))
B <- apply(B, 1, any)
fasta <- c()
for (i in 1:length(B_raw$FASTA)){
  tmp <- strsplit(B_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
B <- as.data.frame(cbind(B, fasta))

# merge data and preapare for plotting
AF_TF_B_comparison  <- merge(B, AF,
                           all = TRUE,
                           by = "fasta")
AF_TF_B_comparison <- merge(AF_TF_B_comparison, TF,
                            all = TRUE,
                            by = "fasta")

AF_TF_B_comparison$B <- as.logical(AF_TF_B_comparison$B)
AF_TF_B_comparison$AF <- as.logical(AF_TF_B_comparison$AF)
AF_TF_B_comparison$TF <- as.logical(AF_TF_B_comparison$TF)
AF_TF_B_comparison <- AF_TF_B_comparison[,-1]*1
AF_TF_B_comparison[AF_TF_B_comparison==0] <- NA

proteins <- as.data.frame(which(!is.na(AF_TF_B_comparison), arr.ind=TRUE))

B <- proteins %>% dplyr::filter(col == 1)
AF <- proteins %>% dplyr::filter(col == 2)
TF <- proteins %>% dplyr::filter(col == 3)


# combine to list
lst <- list("E4_D1_B"=as.double(B[,1]),
            "E4_D1_AF"=as.double(AF[,1]),
            "E4_D1_TF"=as.double(TF[,1]))

# plot results as venn diagram
euler_venn_3comp(lst, 
                 label_venn = c("D1-electrophoretic fractionation","D1-low MW SEC fractionation","D1-high MW SEC fractionation"),
                 color_venn = c(col_X_light, col_X_lowMW, col_X_highMW), 
                 cex_venn = c(1,
                              0.0, 
                              0.0, 
                              0.7,
                              0.7,
                              0.0,
                              1),
                 h_just = 0.5,
                 v_just = 0.5)

# print summary results
print(c("electrophoretic separation","low MW SEC","high MW SEC"))
sum(AF_TF_B_comparison$B,na.rm = TRUE)
sum(AF_TF_B_comparison$AF,na.rm = TRUE)
sum(AF_TF_B_comparison$TF,na.rm = TRUE)
```

## B - Compare D1 fractionation and pool

```{r, fig.width=4,fig.height=4}
# load and prepare data D1 fractionation 
B_raw <- readRDS(file = "results/results_run1_mqpar_extracts_2gether_E4_D1_B_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
B_raw <- B_raw[[1]]
B <- dplyr::select(B_raw, contains("Pool_G"))
B <- apply(B, 1, any)
fasta <- c()
for (i in 1:length(B_raw$FASTA)){
  tmp <- strsplit(B_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
B <- as.data.frame(cbind(B, fasta))

# load and prepare data from D1 pool
aj_raw <- readRDS(file = "results/results_run1_mqpar_extracts_2gether_E4_D1_a-j_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
aj_raw <- aj_raw[[1]]
aj <- dplyr::select(aj_raw, contains("Pool"))
aj <- apply(aj, 1, any)
fasta <- c()
for (i in 1:length(aj_raw$FASTA)){
  tmp <- strsplit(aj_raw$FASTA[i],split='\\|')[[1]][2]
  fasta <- c(fasta,tmp)
}
aj <- as.data.frame(cbind(aj, fasta))

# merge data D1 fractionation and pool and prepare data for plotting
B_aj_comparison  <- merge(B, aj,
                             all = TRUE,
                             by = "fasta")

B_aj_comparison$B <- as.logical(B_aj_comparison$B)
B_aj_comparison$aj <- as.logical(B_aj_comparison$aj)
B_aj_comparison <- B_aj_comparison[,-1]*1
B_aj_comparison[B_aj_comparison==0] <- NA

proteins <- as.data.frame(which(!is.na(B_aj_comparison), arr.ind=TRUE))

B_inter <- proteins %>% dplyr::filter(col == 1)
aj_inter <- proteins %>% dplyr::filter(col == 2)

# combine to list
lst <- list("D1_fractions"=as.double(B_inter[,1]),
            "D1_pool"=as.double(aj_inter[,1]))

# plot results
euler_venn_2comp(lst,
                 label_venn = c("D1-electrophoretic fractionation","D1-total protein extract"),
                 color_venn = c(col_X_light, col_X_dark),
                 cex_venn = c(1,0,1),
                 h_just = 0.5,
                 v_just = 0.5)

# print summary results
print(c("D1 fractionation","D1 pool"))
sum(B_aj_comparison$B,na.rm = TRUE)
sum(B_aj_comparison$aj,na.rm = TRUE)
```


## C - Compare D2 fractionation and pool

```{r, fig.width=4,fig.height=4}
# load and prepare data D2 fractionation and pool
data <- readRDS(file = "results/results_run1_mqpar_extracts_2gether_D2_LFQ_combined_txt_proteinGroups.txt_Unique.RDS")
data <- data[[1]]
data_D2_pool <- dplyr::select(data, contains("Gesamt"))
data_D2_pool <- apply(data_D2_pool, 1, any)
data_D2_fractions <- dplyr::select(data, contains("20"))
data_D2_fractions <- apply(data_D2_fractions, 1, any)
data_D2 <- as.data.frame(cbind(data_D2_pool,data_D2_fractions))
data_D2 <- data_D2*1
data_D2[data_D2==0] <- NA

proteins <- as.data.frame(which(!is.na(data_D2), arr.ind=TRUE))

D2_pool <- proteins %>% dplyr::filter(col == 1)
D2_fractions <- proteins %>% dplyr::filter(col == 2)

# combine to list
lst <- list(D2_fractions=as.double(D2_fractions[,1]),
            D2_pool=as.double(D2_pool[,1]))

# plot results
euler_venn_2comp(lst,
                 label_venn = c("D2-electrophoretic fractionation","D2-total protein extract"),
                 color_venn = c( col_Y_light,col_Y_dark),
                 cex_venn = c(1,0.0,1),
                 h_just = 0.5,
                 v_just = 0.5)
# print summary results
print(c("D2 fractionation","D2 pool"))
sum(data_D2$data_D2_fractions,na.rm = TRUE)
sum(data_D2$data_D2_pool,na.rm = TRUE)
```



# Figure 5 - Electrophoretic separation 

## D1

```{r, fig.width=4,fig.height=4}
# read in data
dat <- as.data.frame(
  cbind(c(126,159,147,241,322,591,661,852,699),
        c(paste("B",1:9,sep=""))))
colnames(dat) <- c("no","fraction")
dat$no <- as.numeric(dat$no)
# re-arrange position
positions <-  c(paste("B",9:1,sep=""))

# create bar-plot
ggplot(data=dat,aes(x=fraction,y=no))+
  geom_bar(stat="identity") +
  coord_flip() +
  geom_text(aes(label=no), hjust=1.5,vjust=0.6, color="white", size=4) +
  theme(axis.title.y = element_blank(),axis.text.y = element_text(size=14)) + 
  scale_y_continuous(breaks = seq(0, 1000, by = 100)) +
  labs(y="Number of proteins per fraction") +
  scale_x_discrete(limits = positions)
```

## D2

```{r, fig.width=4,fig.height=4}
# read in data
dat <- as.data.frame(
  cbind(c(107,241,231,142,145,148,139,166,180,98),
        c(paste("B",1:10,sep=""))))
colnames(dat) <- c("no","fraction")
dat$no <- as.numeric(dat$no)
# re-arrange position
positions <-  c(paste("B",10:1,sep=""))

#create bar-plot
ggplot(data=dat, aes(x=fraction,y=no))+
  geom_bar(stat="identity") +
  coord_flip() +
  geom_text(aes(label=no), hjust=1.5,vjust=0.6, color="white", size=4) +
  theme(axis.title.y = element_blank(),axis.text.y = element_text(size=14)) + 
  scale_y_continuous(breaks = seq(0, 1000, by = 100)) +
  scale_x_discrete(limits = positions) +
  labs(y="Number of proteins per fraction")
```

# Figure 6 - Biological annotation

## Preparation

```{r}
# prepare vectors
results_comp <- c()
length_comp <- c()
results_comp_cc <- c()
length_comp_cc <- c()
results_comp_bp <- c()
length_comp_bp <- c()
results_comp_bp <- c()
name_comp <- c()
name_comp_cc <- c()
name_comp_bp <- c()
```

## Get data

### Get data from Figure 3C

```{r}
# get data D1/D2 comparison
#D1_uni <- GetProteinGOInfo(D1)
#saveRDS(D1_uni,"./go_terms/graph2_D1_uni.RDS")
GO_1 <- readRDS("./go_terms/graph2_D1_uni.RDS")
#D2_uni <- GetProteinGOInfo(D2)
#saveRDS(D2_uni,"./go_terms/graph2_D2_uni.RDS")
GO_2 <- readRDS("./go_terms/graph2_D2_uni.RDS")

################################# molecular function
t_BP <- as.data.frame(table(GO_1$Gene.Ontology..molecular.function.))
t_BP_sorted_1 <- arrange(t_BP, desc(Freq))
t_BP <- as.data.frame(table(GO_2$Gene.Ontology..molecular.function.))
t_BP_sorted_2 <- arrange(t_BP, desc(Freq))
BP <- merge(t_BP_sorted_1, t_BP_sorted_2,
            all = TRUE,
            by = "Var1")
tmp <- c(sum(complete.cases(BP[,-1])),
                  sum(is.na(BP[,3])),
                  sum(is.na(BP[,2])))
results_comp <- c(results_comp, tmp/(length(BP[,1])))
name_comp <- c(name_comp,"graph2A-D1+D2","graph2A-onlyD1","graph2A-onlyD2")

################################# cellular.component
t_BP <- as.data.frame(table(GO_1$Gene.Ontology..cellular.component.))
t_BP_sorted_1 <- arrange(t_BP, desc(Freq))
t_BP <- as.data.frame(table(GO_2$Gene.Ontology..cellular.component.))
t_BP_sorted_2 <- arrange(t_BP, desc(Freq))
BP <- merge(t_BP_sorted_1, t_BP_sorted_2,
            all = TRUE,
            by = "Var1")
colnames(BP) <- c("Function", "D1","D2")
tmp <- c(sum(complete.cases(BP[,-1])),
                  sum(is.na(BP[,3])),
                  sum(is.na(BP[,2])))
results_comp_cc <- c(results_comp_cc, tmp/(length(BP[,1])))
name_comp_cc <- c(name_comp_cc, "graph2A-D1+D2","graph2A-onlyD1","graph2A-onlyD2")

################################# biological process
t_BP <- as.data.frame(table(GO_1$Gene.Ontology..biological.process.))
t_BP_sorted_1 <- arrange(t_BP, desc(Freq))
t_BP <- as.data.frame(table(GO_2$Gene.Ontology..biological.process.))
t_BP_sorted_2 <- arrange(t_BP, desc(Freq))
BP <- merge(t_BP_sorted_1, t_BP_sorted_2,
            all = TRUE,
            by = "Var1")
colnames(BP) <- c("Function", "D1","D2")
tmp <- c(sum(complete.cases(BP[,-1])),
                  sum(is.na(BP[,3])),
                  sum(is.na(BP[,2])))
results_comp_bp <- c(results_comp_bp, tmp/(length(BP[,1])))
name_comp_bp <- c(name_comp_bp, "graph2A-D1+D2","graph2A-onlyD1","graph2A-onlyD2")
```

### Get data from Figure 3A

```{r, fig.width=4,fig.height=4}
GO_1 <- readRDS("./go_terms/graph2_E4_uni.RDS")
GO_2 <- readRDS("./go_terms/graph2_E5_uni.RDS")

################################# molecular function
t_BP <- as.data.frame(table(GO_1$Gene.Ontology..molecular.function.))
t_BP_sorted_1 <- arrange(t_BP, desc(Freq))
t_BP <- as.data.frame(table(GO_2$Gene.Ontology..molecular.function.))
t_BP_sorted_2 <- arrange(t_BP, desc(Freq))
BP <- merge(t_BP_sorted_1, t_BP_sorted_2,
            all = TRUE,
            by = "Var1")
tmp <- c(sum(complete.cases(BP[,-1])),
                  sum(is.na(BP[,3])),
                  sum(is.na(BP[,2])))
results_comp <- c(results_comp, tmp/(length(BP[,1])))
name_comp <- c(name_comp,"graph2C-2D+3D","graph2C-only2D","graph2C-only3D")


################################# cellular.component
t_BP <- as.data.frame(table(GO_1$Gene.Ontology..cellular.component.))
t_BP_sorted_1 <- arrange(t_BP, desc(Freq))
t_BP <- as.data.frame(table(GO_2$Gene.Ontology..cellular.component.))
t_BP_sorted_2 <- arrange(t_BP, desc(Freq))
#t_BP <- as.data.frame(table(GO_3$Gene.Ontology..cellular.component.))
#t_BP_sorted <- arrange(t_BP, desc(Freq))
#t_BP_sorted_3 <- t_BP_sorted[c(1:20),]
BP <- merge(t_BP_sorted_1, t_BP_sorted_2,
            all = TRUE,
            by = "Var1")
#BP <- merge(BP, t_BP_sorted_3, all = TRUE, by = "Var1")
tmp <- c(sum(complete.cases(BP[,-1])),
                  sum(is.na(BP[,3])),
                  sum(is.na(BP[,2])))
results_comp_cc <- c(results_comp_cc, tmp/(length(BP[,1])))
name_comp_cc <- c(name_comp_cc,"graph2C-2D+3D","graph2C-only2D","graph2C-only3D")


################################# biological process
t_BP <- as.data.frame(table(GO_1$Gene.Ontology..biological.process.))
t_BP_sorted_1 <- arrange(t_BP, desc(Freq))
t_BP <- as.data.frame(table(GO_2$Gene.Ontology..biological.process.))
t_BP_sorted_2 <- arrange(t_BP, desc(Freq))
BP <- merge(t_BP_sorted_1, t_BP_sorted_2,
            all = TRUE,
            by = "Var1")
tmp <- c(sum(complete.cases(BP[,-1])),
                  sum(is.na(BP[,3])),
                  sum(is.na(BP[,2])))
results_comp_bp <- c(results_comp_bp, tmp/(length(BP[,1])))
name_comp_bp <- c(name_comp_bp,"graph2C-2D+3D","graph2C-only2D","graph2C-only3D")
      
```

### Get data from Figure 4B

```{r}
GO_1 <- readRDS("./go_terms/graph3_B_B_uni.RDS")
GO_2 <- readRDS("./go_terms/graph3_B_aj_uni.RDS")

################################# molecular function
t_BP <- as.data.frame(table(GO_1$Gene.Ontology..molecular.function.))
t_BP_sorted_1 <- arrange(t_BP, desc(Freq))
t_BP <- as.data.frame(table(GO_2$Gene.Ontology..molecular.function.))
t_BP_sorted_2 <- arrange(t_BP, desc(Freq))
BP <- merge(t_BP_sorted_1, t_BP_sorted_2,
            all = TRUE,
            by = "Var1")
tmp <- c(sum(complete.cases(BP[,-1])),
                  sum(is.na(BP[,3])),
                  sum(is.na(BP[,2])))
results_comp <- c(results_comp, tmp/(length(BP[,1])))
name_comp <- c(name_comp,"graph3B-pool+fracD1","graph3B-onlyFracD1","graph3B-onlyPoolD1")
colnames(BP) <- c("Function", "D1","D1 pool")

################################# cellular compartmnet
t_BP <- as.data.frame(table(GO_1$Gene.Ontology..cellular.component.))
t_BP_sorted_1 <- arrange(t_BP, desc(Freq))
t_BP <- as.data.frame(table(GO_2$Gene.Ontology..cellular.component.))
t_BP_sorted_2 <- arrange(t_BP, desc(Freq))
BP <- merge(t_BP_sorted_1, t_BP_sorted_2,
            all = TRUE,
            by = "Var1")
tmp <- c(sum(complete.cases(BP[,-1])),
                  sum(is.na(BP[,3])),
                  sum(is.na(BP[,2])))
results_comp_cc <- c(results_comp_cc, tmp/(length(BP[,1])))
name_comp_cc <- c(name_comp_cc,"graph3B-pool+fracD1","graph3B-onlyFracD1","graph3B-onlyPoolD1")


################################# biological process
t_BP <- as.data.frame(table(GO_1$Gene.Ontology..biological.process.))
t_BP_sorted_1 <- arrange(t_BP, desc(Freq))
t_BP <- as.data.frame(table(GO_2$Gene.Ontology..biological.process.))
t_BP_sorted_2 <- arrange(t_BP, desc(Freq))
BP <- merge(t_BP_sorted_1, t_BP_sorted_2,
            all = TRUE,
            by = "Var1")
tmp <- c(sum(complete.cases(BP[,-1])),
                  sum(is.na(BP[,3])),
                  sum(is.na(BP[,2])))
results_comp_bp <- c(results_comp_bp, tmp/(length(BP[,1])))
name_comp_bp <- c(name_comp_bp,"graph3B-pool+fracD1","graph3B-onlyFracD1","graph3B-onlyPoolD1")

```


### Get Data from Figure 4C

```{r, fig.width=4,fig.height=4}
GO1<- readRDS("./go_terms/graph3_C_pool_uni.RDS")
GO_2 <- readRDS("./go_terms/graph3_C_fractions_uni.RDS")

################################# molecular function
t_BP <- as.data.frame(table(GO_1$Gene.Ontology..molecular.function.))
t_BP_sorted_1 <- arrange(t_BP, desc(Freq))
t_BP <- as.data.frame(table(GO_2$Gene.Ontology..molecular.function.))
t_BP_sorted_2 <- arrange(t_BP, desc(Freq))
BP <- merge(t_BP_sorted_1, t_BP_sorted_2,
            all = TRUE,
            by = "Var1")
tmp <- c(sum(complete.cases(BP[,-1])),
                  sum(is.na(BP[,3])),
                  sum(is.na(BP[,2])))
results_comp <- c(results_comp, tmp/(length(BP[,1])))
name_comp <- c(name_comp, "graph3C-pool+fracD2","graph3C-onlyFracD2","graph3C-onlyPoolD2")
colnames(BP) <- c("Function", "D2","D2pool")

################################# cellular.component
t_BP <- as.data.frame(table(GO_1$Gene.Ontology..cellular.component.))
t_BP_sorted_1 <- arrange(t_BP, desc(Freq))
t_BP <- as.data.frame(table(GO_2$Gene.Ontology..cellular.component.))
t_BP_sorted_2 <- arrange(t_BP, desc(Freq))
BP <- merge(t_BP_sorted_1, t_BP_sorted_2,
            all = TRUE,
            by = "Var1")
tmp <- c(sum(complete.cases(BP[,-1])),
                  sum(is.na(BP[,3])),
                  sum(is.na(BP[,2])))
results_comp_cc <- c(results_comp_cc, tmp/(length(BP[,1])))
name_comp_cc <- c(name_comp_cc,"graph3C-pool+fracD2","graph3C-onlyFracD2","graph3C-onlyPoolD2")

################################# biological process
t_BP <- as.data.frame(table(GO_1$Gene.Ontology..biological.process.))
t_BP_sorted_1 <- arrange(t_BP, desc(Freq))
t_BP <- as.data.frame(table(GO_2$Gene.Ontology..biological.process.))
t_BP_sorted_2 <- arrange(t_BP, desc(Freq))
BP <- merge(t_BP_sorted_1, t_BP_sorted_2,
            all = TRUE,
            by = "Var1")
tmp <- c(sum(complete.cases(BP[,-1])),
                  sum(is.na(BP[,3])),
                  sum(is.na(BP[,2])))
results_comp_bp <- c(results_comp_bp, tmp/(length(BP[,1])))
name_comp_bp <- c(name_comp_bp,"graph3C-pool+fracD2","graph3C-onlyFracD2","graph3C-onlyPoolD2")
```


## Plot data


```{r fig.height=10, fig.width=10, message=FALSE, warning=FALSE}
# define additional colors  
col_D1D2 <- rgb(160,0,108, maxColorValue = 255)
col_D1frac_pool <- rgb(231,63,52, maxColorValue = 255)
col_D2frac_pool <- rgb(67,59,229, maxColorValue = 255)

# define colors for levels
custom_colors <- c(
  "B" = col_D1D2,
  "A" = col_X_dark,
  "C" = col_Y_dark,
  "E" = "#623E8D",
  "D" = "#BE68CE",
  "F" = "#EDA9BA",
  "H" = col_D1frac_pool,
  "G" = col_X_light,
  "I" = col_X_dark,
  "K" = col_D2frac_pool,
  "J" = col_Y_light,
  "L" = col_Y_dark)

# create vector for comparisons
tmp <- c(rep("D1 vs. D2",3),
         rep("2D vs. 3D",3),
         rep("D1: frac vs. total",3),
         rep("D2: frac vs. total",3))

# create data.frame for plotting
df <- data.frame(
  Category = rep(tmp, times = 3),
  Type = c(rep(c("biological process", "molecular function", "cellular compartment"), each = 12)),
  Percentage = c(round(results_comp_bp*100,1),
                 round(results_comp*100,1),
                 round(results_comp_cc*100,1)),
  Variable = rep(c("B", "A", "C",
                   "E", "D", "F",
                   "H", "G", "I",
                   "K", "J", "L"),3)
  )

# reorder category
df$Category <- factor(df$Category, levels = c("2D vs. 3D", "D1 vs. D2", "D1: frac vs. total", "D2: frac vs. total"))

# vector for desired order
desired_order <- c("D", "A", "G", "J",
                   "E", "B", "H", "K",
                   "F", "C", "I", "L")

# vector for the labels
desired_order_2 <- c("B"="D1+D2", "A"="D1", "C"="D2",
                     "E"="2D+3D", "D"="2D", "F"="3D",
                     "H"="D1 electrophoretic frac+total", "G"="D1 electrophoretic frac", "I"="D1 total (< 3.5 %)",
                     "K"="D2 electrophoretic frac+total", "J"="D2 electrophoretic frac", "L"="D2 total")

# stacked barplot
p <- ggplot(df, aes(x = Category, y = Percentage, fill = Variable)) +
  geom_bar(stat = "identity", position = "stack",
           color = "white", linewidth = 1) +
  facet_grid(Type ~ ., scales = "free_y", space = "free_y") +
  theme_minimal() +
  labs(x = "",
       y = "Percentage",
       fill = "") + 
  theme(legend.position = "bottom", legend.box = "horizontal",
        legend.key.size = unit(0.4, "cm"),
        legend.background = element_rect(size=0.5, linetype="solid"),
        axis.text.x = element_text(angle = 0, hjust = 0.5),
        legend.text = element_text(size = 8),
        strip.text = element_text(size = 15, margin = margin(20, unit = "pt")),
        text = element_text(size=20),
        strip.clip = "off") +
  guides(fill=guide_legend(nrow=3, byrow=TRUE)) +
  scale_fill_manual(values = custom_colors, breaks = desired_order, labels = desired_order_2) +
  geom_bar_text(aes(label = paste0(Percentage,"%")),
                position = "stack",
                reflow = FALSE,
                min.size = 0.1,
                place = "middle",
                outside = TRUE,
                contrast = TRUE)
p + theme(plot.margin = margin(1, 1, 1, 1, "cm"))

```


# Figure 7 - Bioinformatic Data Analysis

```{r}
# read in data for bioinformatic data analysis
data <- read.csv("results_2gether.csv", sep=";")
```


## A - Razor/Unique vs. MBR

```{r, fig.width=4,fig.height=3}
# calculate ratios
MBR <- (data$Razor.MBR.2.peptides / data$Unique.MBR.2.peptides) * 100
noMBR <- (data$Razor.noMBR.2.peptides / data$Unique.no.MBR.2.peptides) * 100

# create a data frame for plotting
df <- data.frame(Group = c(rep("MBR", length(MBR)), rep("noMBR", length(noMBR))),
                 Deviation = c(MBR, noMBR))

# create the boxplot using ggplot
ggplot(df, aes(x = Group, y = Deviation)) +
  geom_boxplot() +
  labs(y = "Deviation Razor-Unique [%]") +
  scale_x_discrete(labels = c("MBR", "noMBR")) +
  theme_minimal() 
```


## B - Razor/Uniqe vs. proteins

```{r, fig.width=4,fig.height=3}
#  calculate ratios and mean
unique <- (data$Unique.MBR.2.peptides / data$Unique.no.MBR.2.peptides) * 100
mean_unique <- (data$Unique.MBR.2.peptides + data$Unique.no.MBR.2.peptides) / 2

# create a data frame for plotting
df <- data.frame(
  MeanUnique = mean_unique,
  Deviation = unique
)

# create the scatter plot using ggplot
ggplot(df, aes(x = MeanUnique, y = Deviation)) +
  geom_point() +
  labs(x = "Number of identified proteins", y = "Deviation MBR [%]") +
  theme_minimal() +
  scale_x_continuous(limits = c(0, 800)) 


```

## C - MBR vs. proteins

```{r, fig.width=4,fig.height=3}
# calculate ratios
MBR <- (data$Razor.MBR.2.peptides / data$Unique.MBR.2.peptides) * 100
noMBR <- (data$Razor.noMBR.2.peptides / data$Unique.no.MBR.2.peptides) * 100

# calculate means
mean_MBR <- (data$Razor.MBR.2.peptides + data$Unique.MBR.2.peptide) / 2
mean_noMBR <- (data$Razor.noMBR.2.peptides + data$Unique.no.MBR.2.peptides) / 2

# create a data frame for plotting
df <- data.frame(
  MeanNoMBR = mean_noMBR,
  Deviation = noMBR
)

# create the scatter plot using ggplot
ggplot(df, aes(x = MeanNoMBR, y = Deviation)) +
  geom_point() +
  labs(x = "Number of identified proteins", y = "Deviation Razor-Unique [%]") +
  theme_minimal()+
  scale_x_continuous(limits = c(0, 800)) 
```

## D - rel. standard deviation vs. proteins

```{r, fig.width=4,fig.height=3}
# group samples and create vectors
sample_group <- c(seq(1,18,by=3),seq(19,162,by=2))
sd_samples <- c()
mean_samples <- c()

# calculate sd and mean for each pair of data
for (i in 1:length(sample_group)) {
  if (i < 78) {
    tmp <- sd(data$Unique.no.MBR.2.peptides[sample_group[i]:(sample_group[i+1]-1)])
    sd_samples <- c(sd_samples, tmp)
    tmp <- mean(data$Unique.no.MBR.2.peptides[sample_group[i]:(sample_group[i+1]-1)])
    mean_samples <- c(mean_samples, tmp)
  } else {
    tmp <- sd(data$Unique.no.MBR.2.peptides[sample_group[i]:length(data$Unique.no.MBR.2.peptides)])
    sd_samples <- c(sd_samples, tmp)
    tmp <- mean(data$Unique.no.MBR.2.peptides[sample_group[i]:length(data$Unique.no.MBR.2.peptides)])
    mean_samples <- c(mean_samples, tmp)
  }
}

# calculate relative standard deviation
rel_sd <- sd_samples / mean_samples * 100

# calculate medians
median_below_200 <- median(rel_sd[which(mean_samples < 200)])
median_above_200 <- median(rel_sd[which(mean_samples > 200)])

# perform t-test
t_test_result <- t.test(rel_sd[which(mean_samples < 200)], rel_sd[which(mean_samples > 200)])

# create a data frame for plotting
df <- data.frame(
  MeanSamples = mean_samples[-31],
  RelSD = sd_samples[-31] / mean_samples[-31] * 100
)

# create the scatter plot using ggplot
ggplot(df, aes(x = MeanSamples, y = RelSD)) +
  geom_point() +
  labs(x = "Number of identified proteins", y = "RSD [%]") +
  theme_minimal() +
  scale_x_continuous(limits = c(0, 800)) 

# print summary median values
print(paste("Median below 200:", median_below_200))
print(paste("Median above 200:", median_above_200))

# output t-test result
print(t_test_result)

```
